test/profilelogincontrol.spec.js

```diff
@@ -0,0 +1,394 @@
+    profileLoginControl = new ProfileLoginControl($elm);
+  });
+
+  it('exists', function () {
+    expect(ProfileLoginControl).to.exist;
+  });
+
+  describe('for handling its data attributes', () => {
+
+    let invalidDataAttributeRoots;
+    let validDataAttributeRoots;
+
+    beforeEach(function () {
+      invalidDataAttributeRoots = [
+        undefined,
+        null,
+        {},
+        {'fish': 'chips'},
+        [],
+        ['fish', 'chips'],
+        1,
+        0,
+        true,
+        false,
+        'there should be no spaces without commas',
+        'any spaces,should only be around commas',
+        'any spaces, should only be around commas',
+        'any spaces ,should only be around commas',
+        'any spaces , should only be around commas',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames?plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames!plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames@plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames%plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames^plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames&plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames*plus-hyphens, commas,andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames)plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames(plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames}plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames{plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames]plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames[plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames.plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames"plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames\'plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames;plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames:plus-hyphens, commas, andSpaces',
+        'shouldOnlyContainCharactersUseableWithinJavaScriptNames/plus-hyphens, commas, andSpaces',
+        'hyphens-should-not-be-at-the-end-of-the-string-',
+        'hyphens-should-not-be-at-the-end-of-a-word-, in-the-string',
+        '-hyphens-should-not-be-at-the-start-of-the-string',
+        'hyphens-should-not-be-at-the-start-of-a, -word-in-the-string'
+      ];
+
+      validDataAttributeRoots = [
```

Comment:
Aim to get a regex fitting the final version of these into the json schema, for extra stringency.